packages/react-dom/src/events/DOMModernPluginEventSystem.js

```diff
@@ -362,11 +399,19 @@ export function listenToReactPropEvent(
   // this React prop event again.
   listenerMap.set(reactPropEvent, null);
   const dependencies = registrationNameDependencies[reactPropEvent];
+  const dependenciesLength = dependencies.length;
+  // If the dependencies length is 1, that means we're not using a polyfill
+  // plugin like ChangeEventPlugin, BeforeInputPlugin, EnterLeavePlugin and
+  // SelectEventPlugin. SimpleEventPlugin always only has a single dependency.
+  // Given this, we know that we never need to apply capture phase event
+  // listeners to anything other than the SimpleEventPlugin.
+  const registrationCapturePhase =
+    isCaptureRegistrationName(reactPropEvent) && dependenciesLength === 1;
 
-  for (let i = 0; i < dependencies.length; i++) {
+  for (let i = 0; i < dependenciesLength; i++) {
     const dependency = dependencies[i];
-    const capture = capturePhaseEvents.has(dependency);
-
+    const capture =
+      capturePhaseEvents.has(dependency) || registrationCapturePhase;
```

Comment:
We already calculated `registrationCapturePhase` so let's reorder so it goes first and we don't always have to check the Set?